src/components/HomePage/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { connect } from 'react-redux';
+import { addCounterAction, decrementAction, incrementAction } from './actions'
+import HomePage from './HomePage'
+import { ICounter } from './types';
+import { createSelector } from "reselect";
+
+const countersSelector = createSelector(
+  [(state: { toJS: () => { counters: ICounter; }; }) => state.toJS().counters],
```

Comment:
Do not need to transform into Immutable values into plain JS. You should use Immutable into React components

Also, always take into account that `toJS` is very difficult task and can lead to performance decrease.